drivers/rs485_seds/work/assembler/implem/default/C/src/assembler.c

```diff
@@ -0,0 +1,166 @@
+    !! the procedures corresponding to the provided interfaces in this  !!
+    !! file. The up-to-date signatures can be found in the .ads file.   !!
+*/
+#include "assembler.h"
+
+#include <assert.h>
+
+#include <FreeRTOS.h>
+#include <semphr.h>
+
+#include <Utils/ByteFifo.h>
+#include <Escaper.h>
+
+__attribute__((section(".sdramMemorySection")))
+static asn1SccUartHwas uart;
+
+__attribute__((section(".sdramMemorySection")))
+static Escaper escaper;
+#define ENCODED_PACKET_BUFFER_SIZE 100
+__attribute__((section(".sdramMemorySection")))
+static uint8_t encodedPacketBuffer[ENCODED_PACKET_BUFFER_SIZE] = {""};
+#define DECODED_PACKET_BUFFER_SIZE 100
+__attribute__((section(".sdramMemorySection")))
+static uint8_t decodedPacketBuffer[DECODED_PACKET_BUFFER_SIZE] = {""};
+
+#define OUTPUT_BUFFER_SIZE 1000
+__attribute__((section(".sdramMemorySection")))
+static uint8_t outputQueueBuffer[OUTPUT_BUFFER_SIZE];
+__attribute__((section(".sdramMemorySection")))
+static ByteFifo outputQueue;
+
+__attribute__((section(".sdramMemorySection")))
+static asn1SccSystemBus busId;
+
+void assembler_startup(void)
+{
+}
+
+asn1SccUartHwas_Instance assembler_convert_conf_asn_devname_to_hwas
+      (asn1SccRS485_SEDS_Device_T devname)
+{
+   switch (devname) {
+   case asn1SccRS485_SEDS_Device_T_uart4:
+      return UartHwas_Instance_uartHwas_Instance_4;
+   case asn1SccRS485_SEDS_Device_T_uart3:
+      return UartHwas_Instance_uartHwas_Instance_3;
+   case asn1SccRS485_SEDS_Device_T_uart2:
+      return UartHwas_Instance_uartHwas_Instance_2;
+   case asn1SccRS485_SEDS_Device_T_uart1:
+      return UartHwas_Instance_uartHwas_Instance_1;
+   case asn1SccRS485_SEDS_Device_T_uart0:
+   default:
+      return UartHwas_Instance_uartHwas_Instance_0;
+   }
+}
+
+asn1SccUartHwas_Instance assembler_convert_conf_asn_baudrate_to_hwas
+      (asn1SccRS485_SEDS_Baudrate_T baudrate)
+{
+   switch (baudrate) {
+   case asn1SccRS485_SEDS_Baudrate_T_b115200:
+      return UartHwas_Baudrate_uartHwas_Baudrate115200;
+   case asn1SccRS485_SEDS_Baudrate_T_b9600:
+   default:
+      return UartHwas_Baudrate_uartHwas_Baudrate9600;
+   }
+}
+
+void assembler_invoke_broker_receive_packet
+      (enum SystemBus bus_id, uint8_t* const data, const size_t data_size)
+{
+   asn1SccReceivedRequestData request_data;
+   request_data.bus_id = bus_id;
+   request_data.message_data = (asn1SccPlatformSpecificPointer) data;
+   request_data.length = data_size;
+   assembler_RI_Receive(&request_data);
+}
+
+void assembler_send_single_byte_to_uarthwas()
+{
+   asn1SccUartHwasInterfaceType_SendByteAsyncCmd_Type sendByteStructure = {
+      .uart = uart
+   };
+
+   bool isBytePresent = ByteFifo_pull(&outputQueue, &sendByteStructure.byteToSend);
+   if (isBytePresent) {
+      assembler_RI_UartHwas_SendByteAsyncCmd_Pi(&sendByteStructure);
+   } else {
+      // No more bytes in queue, send acknowledgment to Arbiter.
+      assembler_RI_ControlAck();
```

Comment:
Can we rename this to transmission ended or data sent?

Reply:
done